icu4j/main/core/src/main/java/com/ibm/icu/util/IslamicCalendar.java

```diff
@@ -582,28 +579,88 @@ private final static boolean civilLeapYear(int year)
         return (14 + 11 * year) % 30 < 11;
     }
 
+    /**
+     * The yearStart function for Islamic Civil calendar.
+     */
+    private long yearStartIslamicCivil(int year) {
```

Comment:
The naming here sounds a little weird to my ears.  Maybe `yearStartForIslamicCivil()`?  (Same for the other two functions.)

Reply:
ok, I decide to refactor everything into different algorithm by adding an internal private interface